base/commands/viridian/viridian_it_test.go

```diff
@@ -110,11 +122,132 @@ func listClusters_InteractiveTest(t *testing.T) {
 				tcx.WriteStdin([]byte("\\viridian list-clusters\n"))
 				tcx.AssertStderrContains("OK")
 			})
-			cid := ensureClusterRunning(ctx, tcx)
+			c := createOrGetClusterWithState(ctx, tcx, "RUNNING")
 			tcx.WithReset(func() {
 				tcx.WriteStdin([]byte("\\viridian list-clusters\n"))
 				tcx.AssertStderrContains("OK")
-				tcx.AssertStdoutContains(cid)
+				tcx.AssertStdoutContains(c.ID)
+			})
+		})
+	})
+}
+
+func createCluster_NonInteractiveTest(t *testing.T) {
+	viridianTester(t, func(ctx context.Context, tcx it.TestContext) {
+		ensureNoClusterRunning(ctx, tcx)
+		clusterName := it.UniqueClusterName()
+		tcx.CLCExecute(ctx, "viridian", "create-cluster", "--verbose", "--name", clusterName)
+		tcx.AssertStderrContains("OK")
+		fields := tcx.AssertStdoutHasRowWithFields("ID", "Name")
+		info := check.MustValue(tcx.Viridian.GetCluster(ctx, fields["ID"]))
+		require.Equal(t, info.Name, clusterName)
+		waitState(ctx, tcx, info.ID, "RUNNING")
+	})
+}
+
+func createCluster_InteractiveTest(t *testing.T) {
+	viridianTester(t, func(ctx context.Context, tcx it.TestContext) {
+		tcx.WithShell(ctx, func(tcx it.TestContext) {
+			ensureNoClusterRunning(ctx, tcx)
+			tcx.WithReset(func() {
+				tcx.WriteStdin([]byte("\\viridian create-cluster --verbose \n"))
+				tcx.AssertStderrContains("OK")
+				fields := tcx.AssertStdoutHasRowWithFields("ID", "Name")
+				info := check.MustValue(tcx.Viridian.GetCluster(ctx, fields["ID"]))
+				waitState(ctx, tcx, info.ID, "RUNNING")
+			})
+		})
+	})
+}
+
+func stopCluster_NonInteractiveTest(t *testing.T) {
+	viridianTester(t, func(ctx context.Context, tcx it.TestContext) {
+		c := createOrGetClusterWithState(ctx, tcx, "RUNNING")
+		tcx.CLCExecute(ctx, "viridian", "stop-cluster", c.ID)
+		tcx.AssertStderrContains("OK")
+		check.Must(waitState(ctx, tcx, c.ID, "STOPPED"))
+	})
+}
+
+func stopCluster_InteractiveTest(t *testing.T) {
+	viridianTester(t, func(ctx context.Context, tcx it.TestContext) {
+		tcx.WithShell(ctx, func(tcx it.TestContext) {
+			tcx.WithReset(func() {
+				c := createOrGetClusterWithState(ctx, tcx, "RUNNING")
+				tcx.WriteStdin([]byte(fmt.Sprintf("\\viridian stop-cluster %s\n", c.Name)))
```

Comment:
You can also use `tcx.WriteStdinf` instead.

Reply:
Ah, nice. I will change all occurrences of that.